packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js

```diff
@@ -228,6 +228,25 @@ class CSharpVisitor {
         return null;
     }
 
+    /**
+     * Visitor design pattern
+     * @param {Field} field - the object being visited
+     * @param {Object} parameters  - the parameter
+     * @return {Object} the result of visiting or null
+     * @private
+     */
+    visitScalarField(field, parameters) {
+        let fieldType;
+        let scalarField = field.getScalarField();
+        if (scalarField.getType() === 'String') {
+            const type = field.getParent().getModelFile().getType(field.getType());
+            if (type.getName() === 'UUID' && type.getNamespace().startsWith('concerto.scalar')) {
+                fieldType = 'UUID';
```

Comment:
safer to assign `fieldType` to `concerto.scalar.UUID` so we don't lose the namespace in `toCSharpType`.